java_tests/src/test/java/org/lsst/sal/SAL_TestTest.java

```diff
@@ -0,0 +1,427 @@
+package org.lsst.sal;
+
+import Test.*;
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Test;
+
+public class SAL_TestTest extends BaseTestCase {
+
+    @Test
+    public void testGetCurrentTime() {
+        salUtils su = new salUtils();
+        // Run the test twice in the unlikely event that the first run
+        // occurs at a leap second transition.
+        int leapSeconds = 0;
+        double measuredSeconds = 0.0;
+        for (int i = 0; i < 2; i++) {
+            leapSeconds = su.getLeapSeconds();
+            measuredSeconds = remote.getCurrentTime() - (System.currentTimeMillis() / 1000.0);
+            if (Math.abs(leapSeconds - measuredSeconds) - 1 < 0.1) {
+                // We are probably at a leap second transition;
+                // sleep and try again.
+                sleep(2);
+                continue;
+            }
+            break;
+        }
+        Assert.assertEquals(leapSeconds, measuredSeconds, 0.001);
+    }
+
+    @Test
+    @Ignore("No astropy equivalent for java.")
+    public void testGetLeapSeconds() {
+    }
```

Comment:
Surely Dave is willing to add getLeapSeconds to Java?

I'd be a lot more comfortable if these issues you are finding in the unit tests get fixed in SAL. Dave usually fixes things quickly and if anything he can fix quickly would be a big win over committing disabled tests.

Reply:
The way Java handles leap seconds probably is such that this cannot be fixed by Dave. Java basically makes a smooth transition by enlarging the time span of a second during a certain period of time so no "additional" second is needed. It is very ugly and bad, but that's the way it is. If you prefer I can remove the test case.

Reply:
This seems very surprising. The value returned by SAL's time function (which is not this one) is the value of CLOCK_TAI. I don't think Java would change that. I don't know how ts_sal computes the value returned by getLeapSeconds, but again it should be easy to get that value through to Java. In any case I've started a discussion on Slack to learn more about this and we can revisit this based on that discussion.